src/access/AccessRestriction.cpp

```diff
@@ -0,0 +1,280 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AccessRestriction.h"
+
+#include <algorithm>
+#include <lib/core/Global.h>
+
+using namespace chip::Platform;
+
+namespace chip {
+namespace Access {
+
+std::vector<SharedPtr<AccessRestriction::Entry>> AccessRestriction::sCommissioningEntries;
+
+CHIP_ERROR AccessRestriction::CreateFabricEntries(const FabricIndex fabricIndex)
+{
+    for (auto & entry : sCommissioningEntries)
+    {
+        CreateEntry(nullptr, *entry, fabricIndex);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void AccessRestriction::AddListener(EntryListener & listener)
+{
+    if (mListeners == nullptr)
```

Comment:
I realize this was probably copied from the ACL code, but we have proper linked list support in lib/support now.

Reply:
Will fix.  Please see https://github.com/project-chip/connectedhomeip/issues/35165